src/DS/rds.c

```diff
@@ -601,6 +602,13 @@ static void ds_rds_add_xccdf_test_results(xmlDocPtr doc, xmlNodePtr reports,
 {
 	xmlNodePtr root_element = xmlDocGetRootElement(xccdf_result_file_doc);
 
+	if (root_element->ns && root_element->ns->href &&
```

Comment:
I know, but if anyone ever uses this passing NULL arg or maybe malformed xmlDoc it might lead to NULL deref. I know it's a corner case, but I'd put some null checking before dereferencing that pointer just to be sure.